src-electron/validation/validation.js

```diff
@@ -144,22 +149,38 @@ function extractIntegerValue(value) {
   if (/^-?\d+$/.test(value)) {
     return parseInt(value)
   } else if (/^[0-9A-F]+$/i.test(value)) {
+    // TODO -- is this supposed to catch hex format?
     return parseInt(value, 16)
   } else {
     return parseInt(value, 16)
   }
 }
-
 function getBoundsInteger(attribute) {
   return {
-    min: extractIntegerValue(attribute.min),
-    max: extractIntegerValue(attribute.max),
+    min: getIntegerValue(attribute.min, attribute.type),
+    max: getIntegerValue(attribute.max, attribute.type),
   }
 }
 
+function getIntegerValue(attribute, type) {
```

Comment:
Use the following format for adding documentation to your functions as a good practice.
```
/** Description
 * 
 * @param {*} param1 
 * @param {*} param2 
 * @returns Integer
 */